### PR TITLE
Appease Windows compiler

### PR DESCRIPTION
**Pull request by nicolasvasilache**

### CODE REVIEW

**Review by ftynse — APPROVED** [review]

File: compiler/src/iree/compiler/Codegen/TransformDialectStrategies/GPU/MatmulTensorCoreStrategy.cpp [hunk 1]
```diff
@@ -124,11 +124,13 @@ static Value buildHoistOutputPaddingOp(ImplicitLocOpBuilder &b, Value variantH,
 
   // Perform a pass of canonicalization cleanups + folding fill + pad into pad
   // by applying `foldTensorSubsets` and `tilingCanonicalization`.
-  iree_compiler::buildCanonicalizationAndEnablingTransforms(
-      b,
-      ApplyPatternsOpPatterns{.foldTensorSubsets = true,
-                              .tilingCanonicalization = true},
```
Comment (ScottTodd) [inline compiler/src/iree/compiler/Codegen/TransformDialectStrategies/GPU/MatmulTensorCoreStrategy.cpp:130]: Thanks for the fixes! I wonder why builds on Linux succeeded...

MSVC logs: https://github.com/openxla/iree/actions/runs/4741863995/jobs/8419465087#step:9:6842
```
MatmulTensorCoreStrategy.cpp(129): error C7555: use of designated initializers requires at least '/std:c++20'
MatmulTensorCoreStrategy.cpp(267): error C7555: use of designated initializers requires at least '/std:c++20'
```

In CMake, we set C++17, not C++20: https://github.com/openxla/iree/blob/479900afe1467cd10f349894f758135e54d487e2/CMakeLists.txt#L21-L27

That should be making it's way to clang, MSVC, etc.

**Comment by ftynse** [comment]
clang only complains in the `-pedantic` mode, and the complaint is only a warning
```
warning: designated initializers are a C++20 extension [-Wc++20-designator]
```
similarly to other backported language extensions.

**Comment by ScottTodd** [comment]
Ah, we could probably opt-in to that clang warning as an error. It's nice when warnings match across the different compilers we support.

**Comment by ScottTodd** [comment]
Sent https://github.com/openxla/iree/pull/13158 to enable that warning as an error

**Pull request merged**